### PR TITLE
Improvements in rounding and rotating R functions

### PR DESCRIPTION
**Pull request by TolisChal**
This PR:
1. Adds the option of fixed seed in random rotation of a polytope.
2. Returns the linear transformation applied in rounding
3. Fixes a bug when the #vertices>20d of a V-polytope

### CODE REVIEW

File: R-proj/src/rotating.cpp [hunk 1]
```diff
@@ -86,13 +90,14 @@ Rcpp::NumericMatrix rotating (Rcpp::Reference P, Rcpp::Nullable<Rcpp::NumericMat
...
-                TransorfMat = rotating < MT > (ZP);
+                TransorfMat = rotating < MT > (ZP, seed2);
             }
             Mat = extractMatPoly(ZP);
             break;
         }
         case 4: {
-            Vpolytope VP1;
+            throw Rcpp::exception("volesti does not support rotation for this representation currently.");
+            /*Vpolytope VP1;
```
Comment (vissarion) [inline R-proj/src/rotating.cpp:100]: is this piece of code going to be used in the future?

File: R-proj/src/rotating.cpp [hunk 2]
```diff
@@ -27,13 +27,15 @@
...
 //' @param T Optional. A rotation matrix.
+//' @param seed Optional. A fixed seed for the random linear map generator.
 //'
 //' @section warning:
 //' Do not use this function.
 //'
 //' @return A matrix that describes the rotated polytope
 // [[Rcpp::export]]
-Rcpp::NumericMatrix rotating (Rcpp::Reference P, Rcpp::Nullable<Rcpp::NumericMatrix> T = R_NilValue){
+Rcpp::NumericMatrix rotating (Rcpp::Reference P, Rcpp::Nullable<Rcpp::NumericMatrix> T = R_NilValue,
```
Comment (vissarion) [inline R-proj/src/rotating.cpp:37]: why not creating two overloads of `rotating`, one with fixed seed and one without ?

File: include/rotating.h [hunk 3]
```diff
@@ -12,14 +12,19 @@
...
+    unsigned rng_seed = std::chrono::system_clock::now().time_since_epoch().count();
+    RNGType rng(rng_seed);
+    if (!std::isnan(seed)) {
+        unsigned rng_seed = seed;
+        rng.seed(rng_seed);
+    }
     boost::random::uniform_real_distribution<> urdist(-1.0, 1.0);
-    unsigned seed = std::chrono::system_clock::now().time_since_epoch().count();
-    RNGType rng(seed);
+    //unsigned seed = std::chrono::system_clock::now().time_since_epoch().count();
```
Comment (vissarion) [inline include/rotating.h:26]: needed?

File: include/rotating.h [hunk 4]
```diff
@@ -12,14 +12,19 @@
 
 
 template <typename MT, typename Polytope>
-MT rotating(Polytope &P){
+MT rotating(Polytope &P, double seed = std::numeric_limits<double>::signaling_NaN()){
```
Comment (vissarion) [inline include/rotating.h:15]: same here, why not two overloads?

File: include/rounding.h [hunk 5]
```diff
@@ -99,27 +99,47 @@ std::pair <NT, NT> rounding_min_ellipsoid(Polytope &P , const std::pair<Point,NT
...
-void get_vpoly_center(Polytope &P) {
-
-    typedef typename Polytope::NT 	NT;
-    typedef typename Polytope::MT 	MT;
-    typedef typename Polytope::VT 	VT;
-    typedef typename Polytope::PolytopePoint 	Point;
+// ----- ROUNDING ------ //
+// main rounding function
+template <typename MT, typename VT, typename Polytope, typename Point, typename Parameters, typename NT>
+std::pair< std::pair<MT, VT>, NT >  rounding_min_ellipsoid(Polytope &P , const std::pair<Point,NT> &InnerBall, const Parameters &var) {
```
Comment (vissarion) [inline include/rounding.h:105]: shorter line?

File: include/rounding.h [hunk 6]
```diff
@@ -135,8 +155,69 @@ void get_vpoly_center(Polytope &P) {
...
+        size_t w=1000;
+        KhachiyanAlgo(Ap,0.01,w,Q,c2); // call Khachiyan algorithm
+
+        //MT E(n,n);
+        VT e(n);
+
+        //Get ellipsoid matrix and center as Eigen objects
+        for(unsigned int i=0; i<n; i++){
+            e(i)=NT(c2(i));
+        //for (unsigned int j=0; j<n; j++){
```
Comment (vissarion) [inline include/rounding.h:214]: needed?

File: include/rounding.h [hunk 7]
```diff
@@ -135,8 +155,69 @@ void get_vpoly_center(Polytope &P) {
...
+            for (int i=0 ; i<rpit->dimension(); i++){
+                Ap(i,j)=double((*rpit)[i]);
+            }
+        }
+        boost::numeric::ublas::matrix<double> Q(n,n);
+        boost::numeric::ublas::vector<double> c2(n);
+        size_t w=1000;
+        KhachiyanAlgo(Ap,0.01,w,Q,c2); // call Khachiyan algorithm
+
+        //MT E(n,n);
```
Comment (vissarion) [inline include/rounding.h:208]: needed?

File: R-proj/src/rotating.cpp (hunk 1 above)
Comment (TolisChal) [reply R-proj/src/rotating.cpp:100]: I removed this part.

File: R-proj/src/rotating.cpp (hunk 2 above)
Comment (TolisChal) [reply R-proj/src/rotating.cpp:37]: We use the same code structure for polytope generators as well.

File: include/rotating.h (hunk 3 above)
Comment (TolisChal) [reply include/rotating.h:26]: removed

File: include/rotating.h (hunk 4 above)
Comment (TolisChal) [reply include/rotating.h:15]: overloaded

File: include/rounding.h (hunk 5 above)
Comment (TolisChal) [reply include/rounding.h:105]: done

File: include/rounding.h (hunk 7 above)
Comment (TolisChal) [reply include/rounding.h:208]: removed

File: include/rounding.h (hunk 6 above)
Comment (TolisChal) [reply include/rounding.h:214]: removed

**Pull request merged**